Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditorSettings.cpp

```diff
@@ -0,0 +1,126 @@
+    }
+
+    DocumentPropertyEditorSettings::~DocumentPropertyEditorSettings()
+    {
+        SaveAndCleanExpanderStates();
+    }
+
+    void DocumentPropertyEditorSettings::SaveExpanderStates()
+    {
+        m_settingsRegistrar.StoreObjectSettings(m_fullSettingsRegistryPath, this);
+
+        // Configuration that the dumper util will use when collecting our data from the SettingsRegistry
+        AZ::SettingsRegistryMergeUtils::DumperSettings dumperSettings;
+        dumperSettings.m_prettifyOutput = true;
+        dumperSettings.m_includeFilter = [pathFilter = m_settingsRegistryBasePath](AZStd::string_view path)
+        {
+            return pathFilter.starts_with(path.substr(0, pathFilter.size()));
+        };
+        dumperSettings.m_jsonPointerPrefix = m_settingsRegistryBasePath;
+
+        auto outcome = m_settingsRegistrar.SaveSettingsToFile(m_fullSettingsFilepath, dumperSettings, m_settingsRegistryBasePath);
+        if (!outcome.IsSuccess())
+        {
+            AZ_Warning("DocumentPropertyEditorSettings", false, outcome.GetError().c_str());
+        }
+    }
+
+    bool DocumentPropertyEditorSettings::LoadExpanderStates()
+    {
+        auto loadOutcome = m_settingsRegistrar.LoadSettingsFromFile(m_fullSettingsFilepath);
+        if (loadOutcome.IsSuccess())
+        {
+            auto getOutcome = m_settingsRegistrar.GetObjectSettings(this, m_fullSettingsRegistryPath);
+            if (!getOutcome.IsSuccess())
+            {
+                AZ_Warning("DocumentPropertyEditorSettings", false, getOutcome.GetError().c_str());
+                return false;
+            }
+        }
+        else
+        {
+            AZ_Warning("DocumentPropertyEditorSettings", false, loadOutcome.GetError().c_str());
+            return false;
+        }
+
+        return true;
+    }
+
+    void DocumentPropertyEditorSettings::SaveAndCleanExpanderStates()
+    {
+        // Attempt to remove old settings from registry if the local state was successfully cleaned
+        // This way we save and dump to file the most accurate expander settings
+        if (m_cleanExpanderStateCallback && m_cleanExpanderStateCallback(m_expandedElementStates))
+        {
+            if (!m_settingsRegistrar.RemoveSettingFromRegistry(m_fullSettingsRegistryPath))
+            {
+                AZ_Warning("DocumentPropertyEditorSettings", false,
+                    "Failed to clean registry state %s before saving", m_fullSettingsRegistryPath.c_str());
+            }
+        }
+
+        if (!m_expandedElementStates.empty())
+        {
+            SaveExpanderStates();
+        }
+    }
+
+    void DocumentPropertyEditorSettings::SetExpanderStateForRow(const AZ::Dom::Path& rowPath, bool isExpanded)
+    {      
+        m_expandedElementStates[rowPath.ToString()] = isExpanded;
+    }
+
+    bool DocumentPropertyEditorSettings::GetExpanderStateForRow(const AZ::Dom::Path& rowPath)
+    {
+        AZStd::string_view strPath = rowPath.ToString();
```

Comment:
This needs to be an `AZStd::string` otherwise the string_view is pointing to a temporary string
```suggestion
        AZStd::string strPath = rowPath.ToString();
```

Reply:
A variable lifetime is tied to a "name" in C/C++.

The dangerous part about string_view is that it is allowed to be the `AZStd::string` class conversion has a conversion operator to convert to string_view which doesn't use reference qualifiers, which makes string_view vulnerable to pointing to dangling reference.

There reason it doesn't take reference qualifiers into account is because this code valid
```
void Func(AZStd::string view);
// ...
Func(AZStd::string{"Hello"});
```

The lifetime of `AZStd::string{"Hello"}` doesn't end until after the statement containing function is complete
https://en.cppreference.com/w/cpp/string/basic_string/operator_basic_string_view